src/Interfaces/AuthorizationProviderInterface.php

```diff
@@ -0,0 +1,12 @@
+<?php
+declare(strict_types=1);
+
+namespace Firehed\API\Interfaces;
+
+interface AuthorizationProviderInterface
+{
+    public function authorize(
```

Comment:
This is somewhat limited since it performs all logic pre-execute. This means that in some situations, the endpoint will need to perform additional logic - namely, [IDOR prevention](https://www.owasp.org/index.php/Insecure_Direct_Object_Reference_Prevention_Cheat_Sheet). Not sure if there's much to do here beyond best practices docs.